cpp/src/arrow/dataset/scanner.cc

```diff
@@ -708,8 +709,12 @@ Result<ProjectionDescr> ProjectionDescr::FromNames(std::vector<std::string> name
   for (size_t i = 0; i < exprs.size(); ++i) {
     exprs[i] = compute::field_ref(names[i]);
   }
+  auto fields = dataset_schema.fields();
+  for (const auto& aug_field : kAugmentedFields) {
+    fields.push_back(aug_field);
+  }
   return ProjectionDescr::FromExpressions(std::move(exprs), std::move(names),
-                                          dataset_schema);
+                                          Schema(fields));
```

Comment:
Ah, I think the python test failure is because we are losing the metadata of the dataset schema here.  So 
```suggestion
                                          Schema(fields, dataset_schema.metadata()));
```

Reply:
Made the change.